crates/goose/src/providers/types/tool.rs

```diff
@@ -1,140 +1,123 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
 use std::fmt::Debug;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ToolParameter {
+    /// Description of what the parameter does
+    pub description: String,
+    /// The type of the parameter (e.g., "string", "number", "boolean")
+    pub parameter_type: String,
+    /// Whether this parameter is required
+    pub required: bool,
+}
+
+impl ToolParameter {
```

Comment:
can this support arrays and objects? i wanna say the following tool would work with chat completion:
```
{
  "parameters": {
    "type": "object",
    "properties": {
      "action": {
        "type": "string",
        "description": "The action to be performed, such as displaying FAQs"
      },
      "faq_entries": {
        "type": "array",
        "description": "A list of FAQ entries, each with a question and an answer",
        "items": {
          "type": "object",
          "properties": {
            "question": {
              "type": "string",
              "description": "The question text for the FAQ entry"
            },
            "answer": {
              "type": "string",
              "description": "The answer text for the FAQ entry"
            }
          },
          "required": ["question", "answer"]
        }
      }
    },
    "required": ["action", "faq_entries"]
  }
}
```

Reply:
yeah good point, let me think about this - might have to give up some of the type safety here and go back to how you had it with just Value